mlflow/deployments/base.py

```diff
@@ -70,6 +70,226 @@ def target_help():
     )
 
 
+class BaseEndpointClient(abc.ABC):
+    """
+        Base class exposing Python model endpoint APIs.
+
+        Plugin implementors should define target-specific logic via a subclass of ``BaseEndpointClient``
+        within the plugin module, and customize method docstrings with target-specific information.
+
+        .. Note::
+            Subclasses should raise :py:class:`mlflow.exceptions.MlflowException` in error cases (e.g.
+            on failure to deploy a model).
+        """
+
+    def __init__(self, target_uri):
+        self.target_uri = target_uri
+
+    @abc.abstractmethod
+    def create_endpoint(self, endpoint_name, config=None):
```

Comment:
`endpoint_name` seems not necessary here. Generic `name` should be sufficient. In any case please update the comments (at the moment it contains both `endpoint_name` and `name`).
Ditto below for `update_endpoint`, `delete_endpoint` and `get_endpoint`

Reply:
I switched to `name` for this method, but left the others with the `endpoint` parameter, as we'd talked about previously. Please let me know if you think it should be handled differently.